packages/fast-usdc/src/utils/address.js

```diff
@@ -0,0 +1,57 @@
+import { makeError, q } from '@endo/errors';
+
+/**
+ * Very minimal 'URL query string'-like parser that handles:
+ *  - Query string delimiter (?)
+ *  - Key-value separator (=)
+ *  - Query parameter separator (&)
+ *
+ * Does not handle:
+ * - Subpaths (`agoric1bech32addr/opt/account?k=v`)
+ * - URI encoding/decoding (`%20` -> ` `)
+ * - Multiple question marks (foo?bar=1?baz=2)
+ * - Empty parameters (foo=)
+ * - Array parameters (`foo?k=v1&k=v2` -> k: [v1, v2])
+ * - Parameters without values (foo&bar=2)
+ */
+export const addressTools = {
+  /**
+   * @param {string} address
+   * @returns {boolean}
+   */
+  hasQueryParams: address => {
+    return address.includes('?');
```

Comment:
`agoric1foo?bar?baz` would return true but is invalid. for this to be true it `getQueryParams` should not error. 
```suggestion
    try {
      addressTools.getQueryParams(address);
      return result.params.length > 0;
    } catch {
      return false;
     }      
```